ui/components/src/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useMemo } from 'react';
+import { UnitOptions, formatValue } from '@perses-dev/core';
+import { use, EChartsCoreOption } from 'echarts/core';
+import { BarChart as EChartsBarChart } from 'echarts/charts';
+import { GridComponent, DatasetComponent, TitleComponent, TooltipComponent } from 'echarts/components';
+import { CanvasRenderer } from 'echarts/renderers';
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+import { EChart } from '../EChart';
+
+use([EChartsBarChart, GridComponent, DatasetComponent, TitleComponent, TooltipComponent, CanvasRenderer]);
+
+export interface BarChartData {
+  label: string;
+  value: number | null | undefined;
+}
+
+export interface BarChartProps {
+  width: number;
+  height: number;
+  data: BarChartData[] | null;
+  unit: UnitOptions;
+}
+
+export function BarChart(props: BarChartProps) {
+  const { width, height, data, unit } = props;
+  const chartsTheme = useChartsTheme();
+
+  const option: EChartsCoreOption = useMemo(() => {
```

Comment:
I had a branch at one point where I started to do this for LineChart but someday I want to start using ComposeOption instead of EChartsCoreOption, not a blocker or anything 

Here is more info about it and tracing started using it internally too

https://apache.github.io/echarts-handbook/en/basics/import/#:~:text=in%20your%20project.-,creating%20an%20option%20type%20in%20typescript,-For%20developers%20who